### PR TITLE
Add attribute `data-*` in some elements, update case accordingly, add new case

### PR DESCRIPTION
**Pull request by henrywang**
Add attribute `data-*` in some elements to make element locating easy.
Update test case according to new changes.
Add new case to cover "Selected Components" tab and "Dependencies" tab.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -114,6 +114,9 @@ test_rpmbuild: buildrpm_image
 test_rpmbuild_cockpit-composer: buildrpm_image
 	sudo docker run --rm --name buildrpm -v `pwd`:/welder welder/buildrpm:latest make cockpit-composer-rpm cockpit-composer-srpm
 
+debug_install:
+	sudo rm -rf /usr/share/cockpit/welder && sudo cp -r `pwd`/public /usr/share/cockpit/welder
```
Comment (martinpitt) [inline Makefile]: IMHO this is wrong, as it scribbles over package manager territory. It's much better to create a ~/.local/share/cockpit/welder symlink to `$(pwd)/public`. Maybe call it "local-install"?

Also, this new target needs to be added to `.PHONY`.

File: components/ListView/ComponentInputs.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 import React from 'react';
-import {defineMessages, injectIntl, intlShape} from 'react-intl';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
 import PropTypes from 'prop-types';
 import ComponentTypeIcons from '../../components/ListView/ComponentTypeIcons';
 
 const messages = defineMessages({
   hideDetails: {
-    defaultMessage: "Hide Details"
+    defaultMessage: 'Hide Details'
```
Comment (martinpitt) [inline components/ListView/ComponentInputs.js]: Note that translatable strings must be marked with `_("...")`, and for xgettext it *has* to be double quotes. This PR doesn't make things any worse, but this goes into the wrong direction.

File: components/ListView/ComponentInputs.js [hunk 3]
```diff
@@ -1,16 +1,16 @@
 /* global $ */
 
 import React from 'react';
-import {defineMessages, injectIntl, intlShape} from 'react-intl';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
```
Comment (martinpitt) [inline components/ListView/ComponentInputs.js]: There are a lot of these formatting changes in this PR. Would be nice if they could be done in a separate commit, as this makes it hard to review what actually changed.

**Comment by henrywang** [comment]
@martinpitt, I deleted makefile target change and will move it to another PR. All of the formatting changes are removed from this PR.
The only failed case is due to [issue 471](https://github.com/weldr/welder-web/issues/471).

**Comment by henrywang** [comment]
@martinpitt, yes. It's in another PR.

**Pull request merged**